src/github.com/getlantern/borda/borda.go

```diff
@@ -0,0 +1,246 @@
+package borda
+
+import (
+	"fmt"
+	"math"
+	"net/http"
+	"time"
+
+	"github.com/getlantern/eventual"
+	"github.com/golang/glog"
+	"github.com/influxdata/influxdb/client/v2"
+)
+
+// Measurement represents a measurement at a point in time. It maps to a "point"
+// in InfluxDB.
+type Measurement struct {
+	// Name is the name of the measurement (e.g. cpu_usage). It maps to the "key"
+	// in "InfluxDB".
+	Name string `json:"name"`
+
+	// Ts records the time of the measurement.
+	Ts time.Time `json:"ts,omitempty"`
+
+	// Values contains numeric values of the measurement. These will be stored as
+	// "fields" in InfluxDB.
+	//
+	// Example: { "num_errors": 67 }
+	Values map[string]float64 `json:"values,omitempty"`
+
+	// Dimensions captures key/value pairs which characterize the measurement.
+	// Dimensions are stored as "tags" or "fields" in InfluxDB depending on which
+	// dimensions have been configured as "IndexedDimensions" on the Collector.
+	//
+	// Example: { "requestid": "18af517b-004f-486c-9978-6cf60be7f1e9",
+	//            "ipv6": "2001:0db8:0a0b:12f0:0000:0000:0000:0001",
+	//            "host": "myhost.mydomain.com",
+	//            "total_cpus": "2",
+	//            "cpu_idle": 10.1,
+	//            "cpu_system": 53.3,
+	//            "cpu_user": 36.6,
+	//            "connected_to_internet": true }
+	Dimensions map[string]interface{} `json:"dimensions,omitempty"`
+}
+
+// WriteFunc is a function that writes a batch to the database
+type WriteFunc func(client.BatchPoints) error
+
+// Collector collects Measurements
+type Collector interface {
+	http.Handler
+
+	// Submit submits a measurement to the Collector
+	Submit(*Measurement)
+
+	// Wait waits up to timeout for the Collector to finish running and returns
+	// the error that caused the Collector to terminate. A timeout of -1 causes
+	// Wait to block indefinitely.
+	Wait(timeout time.Duration) error
```

Comment:
Actually, now that I look at it, we're okay.  There are only two reasons this will terminate:

1. Unable to create a batch.  This is not a transient error and we can't recover from it.  In practice, it should never happen.

2. Exhausted allowed retries to commit a batch.  This is a serious condition meaning that we're giving up.  We can set the retries appropriately high.

So, I won't change anything here.